interfaces/builtin/hardware_control.go

```diff
@@ -0,0 +1,51 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const hardwareControlSummary = `allows control of system hardware`
+
+const hardwareControlBaseDeclarationSlots = `
+  hardware-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hardwareControlConnectedPlugAppArmor = `
+# Description: This interface allows for controlling hardware and platform
+# devices connected to the system.
+# This is reserved because it allows potentially disruptive operations and
+# access to devices which may contain sensitive information.
+
+# files in /sys pertaining to hardware (eg, 'lspci -A linux-sysfs')
+/sys/{block,bus,class,devices,firmware}/{,**} rw,
```

Comment:
Is `rw` sufficient or do you also need `k` (locking)?

Reply:
I haven't had any locking requests come up as denied in my testing, rw seems sufficient - but my testing has been limited currently to the PWM devices that exposed the need for such an interface.